lib/Sema/CSSimplify.cpp

```diff
@@ -8533,6 +8533,21 @@ bool ConstraintSystem::simplifyAppliedOverloadsImpl(
     ConstraintLocatorBuilder locator) {
   if (shouldAttemptFixes()) {
     auto arguments = argFnType->getParams();
+
+    // Don't attempt to filter overloads when solving for code completion
+    // because presence of code completion token means that this call
+    // could be malformed e.g. missing arguments after the token.
+    if (llvm::any_of(arguments, [](const AnyFunctionType::Param &argument) {
+          if (auto *typeVar =
+                  argument.getPlainType()->getAs<TypeVariableType>()) {
+            if (auto *locator = typeVar->getImpl().getLocator())
+              return locator->directlyAt<CodeCompletionExpr>();
```

Comment:
@nathawes Just to make sure that I did the right thing - we do want to avoid filtering only overloads where argument is a code completion token right? I think it's okay to filter other calls present in the expression to avoid any performance regressions, that shouldn't affect completion results right?

Reply:
Looks like I have been too optimistic, any call involving code completion token would be under-defined, I'm going to relax this check to just `isForCodeCompletion()` to cover that.